pkg/ingester/ingester.go

```diff
@@ -399,6 +405,19 @@ func New(cfg Config, limits *validation.Overrides, ingestersRing ring.ReadRing,
 	i.compactionIdleTimeout = util.DurationWithPositiveJitter(i.cfg.BlocksStorageConfig.TSDB.HeadCompactionIdleTimeout, compactionIdleTimeoutJitter)
 	level.Info(i.logger).Log("msg", "TSDB idle compaction timeout set", "timeout", i.compactionIdleTimeout)
 
+	if ingestCfg := cfg.IngestStorageConfig; ingestCfg.Enabled {
+		kafkaCfg := ingestCfg.KafkaConfig
+
+		partitionID, err := ingest.IngesterPartition(cfg.IngesterRing.InstanceID)
+		if err != nil {
+			return nil, errors.Wrap(err, "calculating ingest storage partition ID")
+		}
+		i.ingestReader, err = ingest.NewReaderForPusher(kafkaCfg.Address, kafkaCfg.Topic, kafkaCfg.ClientID, partitionID, i, log.With(logger, "component", "ingest_reader"), registerer)
```

Comment:
[nit] Not a blocker, but I would pass the whole `kafkaCfg` like we do for the writer.

Reply:
ah, right. I had this change stashed even